roles/gateway/templates/auth-openvpn

```diff
@@ -36,16 +36,16 @@ def authenticate():
     #
     # We'll just redo that quickly for usernames, to ensure they are safe.
 
-    user = re.sub(r'[^a-zA-Z0-9_.@-]', '_', user)
+    user = re.sub(rb'[^a-zA-Z0-9_.@-]', '_', user)
 
     def find_user(wanted):
-        with file('{{ openvpn_data_dir }}/users') as f:
+        with open('/etc/openvpn/cephco-sepia.data/users', 'rb') as f:
             for line in f:
-                assert line.endswith('\n')
+                assert line.endswith(b'\n')
                 line = line[:-1]
-                if line.startswith("#") or len(line) == 0:
+                if line.startswith(b'#') or len(line) == 0:
                     continue
-                (username, salt, correct) = line.split(' ', 2)
+                (username, salt, correct) = line.split(b' ', 2)
```

Comment:
github won't let me comment on line 54 because it's not near enough to a line that was modified in this PR, but `salt` will need to be `bytes` to prevent `inner.update(salt)` from throwing a `TypeError`

Reply:
I swear I tested this, but I don't know how you're not right.  Looking.

Reply:
actually, no, changing the split string to b' ' means that the outputs are also bytes.

Reply:
The issue is with the line:
`salt = 'not-found'`
because:
```
>>> inner = hashlib.new('sha256')
>>> salt = 'not-found'
>>> inner.update(salt)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Strings must be encoded before hashing
```

Reply:
Got that and the won't-match 'correct' string too.